### ISSUE TITLE
Received 6D00 response to select file

### ISSUE THREAD

**Issue by bertrandmartel** [martinpaljak/GlobalPlatformPro#76]
My question is not so much about Global Platform Pro but about a `6D00` response on file select command . I've built the following [HellowWorld applet](http://javacard.vetilles.com/2006/09/17/hello-world-smart-card/) using [ant-javacard](https://github.com/martinpaljak/ant-javacard) : 

```java
package com.vetilles.helloworld;

import javacard.framework.*;

public class HelloWorld extends Applet {

    private final static byte[] hello = {0x48, 0x65, 0x6c, 0x6c, 0x6f};

    public static void install(byte[] buffer, short offset, byte length) {
        (new HelloWorld()).register();
    }

    public void process(APDU apdu) {
        byte[] buf = apdu.getBuffer();

        switch (buf[ISO7816.OFFSET_INS]) {
            case 0x40:
                Util.arrayCopy(hello, (byte) 0, buf, ISO7816.OFFSET_CDATA, (byte) 5);
                apdu.setOutgoingAndSend(
                        ISO7816.OFFSET_CDATA, (byte) 5);
                break;
            default:
                ISOException.throwIt(ISO7816.SW_INS_NOT_SUPPORTED);
        }
    }
}
```
Build is ok, applet is installed : 

```
[main] WARN pro.javacard.gp.GlobalPlatform - GET STATUS failed for 80F21000024F0000 with 6a81
# Mode: GP211
ISD: A000000003000000 (INITIALIZED)
     Privs:   SecurityDomain, CardLock, CardTerminate, CardReset, CVMManagement

APP: 01020304050607080901 (SELECTABLE)

PKG: 010203040506070809 (LOADED)
```
When I select the AID and send data, it returns correct output `48656C6C6F 9000` using the following command : 

```
gp -dv -a 00A404000A0102030405060708090100 -a 0040000000
```

it gives 
```
[DEBUG] PlaintextKeys - static keys: 
ENC: Ver:0 ID:0 Type:DES3 Len:16 Value:404142434445464748494A4B4C4D4E4F KCV: 8BAF47
MAC: Ver:0 ID:0 Type:DES3 Len:16 Value:404142434445464748494A4B4C4D4E4F KCV: 8BAF47
KEK: Ver:0 ID:0 Type:DES3 Len:16 Value:404142434445464748494A4B4C4D4E4F KCV: 8BAF47
# Detected readers from JNA2PCSC
[ ] Gemalto Prox Dual USB PC Link Reader [Prox-DU Contactless_10800061] 00 00
[*] Gemalto Prox Dual USB PC Link Reader [Prox-DU Contact_10800061] 01 00
SCardConnect("Gemalto Prox Dual USB PC Link Reader [Prox-DU Contact_10800061] 01 00", T=*) -> T=1, 3BF71800008031FE45736674652D6E66C4
SCardBeginTransaction("Gemalto Prox Dual USB PC Link Reader [Prox-DU Contact_10800061] 01 00")
Reader: Gemalto Prox Dual USB PC Link Reader [Prox-DU Contact_10800061] 01 00
ATR: 3BF71800008031FE45736674652D6E66C4
More information about your card:
    http://smartcard-atr.appspot.com/parse?ATR=3BF71800008031FE45736674652D6E66C4

A>> T=1 (4+0010) 00A40400 0A 01020304050607080901 00
A<< (0000+2) (31ms) 6D00
A>> T=1 (4+0000) 00400000 00 
A<< (0005+2) (12ms) 48656C6C6F 9000
A>> T=1 (4+0000) 00A40400 00 
A<< (0018+2) (52ms) 6F108408A000000003000000A5049F6501FF 9000
[DEBUG] GlobalPlatform - Auto-detected ISD AID: A000000003000000
[DEBUG] GlobalPlatform - Auto-detected block size: 255
SCardEndTransaction()
SCardDisconnect("Gemalto Prox Dual USB PC Link Reader [Prox-DU Contact_10800061] 01 00", true)
```
But notice that the response of AID select `00A40400 0A 01020304050607080901 00` is `6D00`, shouldn't be `9000` ? I suspect something is wrong somewhere. Looking at [this documentation](http://www.cardwerk.com/smartcards/smartcard_standard_ISO7816-4_6_basic_interindustry_commands.aspx#chap6_11), I didn't find anything explaining why this would happen.

But even if it returns `6D00`, the select operation apparently did work. Do you know what may be the issue ? (or maybe there's no issue at all but `6D00` seems weird to me?)

Also, is `gp -dv -a <some apdu> -a <some more apdu>`, the recommended way to use global platform pro to send apdu ? I tried to use `gp --sdaid <AID> -a <some apdu>`, but the apdu got sent before the select command is sent. Maybe there is an easier way ?

Thanks a lot

**Comment by bertrandmartel** [comment]
It was a misunderstanding of how `process(APDU apdu)` was called. Sorry for bothering you

**Comment by martinpaljak** [comment]
Too bad, I was hoping for a great bug report :)

-a is for sending "plaintext APDU-s", which must include any SELECT-s (or any other structures known by card). Maybe this should be clarified in README

**Comment by johnnytshi** [comment]
I am new to this, apologies for asking dumb question, what is the correct way to issue the above cmd?

**Comment by martinpaljak** [comment]
Which command?

**Comment by johnnytshi** [comment]
I guess I am confused about the SELECT 
> -a is for sending "plaintext APDU-s", which must include any SELECT-s
Does that mean I always have to issue SELECT cmd together with my "actual" cmd? Like the e.g. above: gp -dv -a 00A404000A0102030405060708090100 -a 0040000000

Thanks!

**Comment by martinpaljak** [comment]
Yes you do. But that is a neat addition - (ab)using --applet for an automagic SELECT before the rest of the APDU-s

**Comment by martinpaljak** [comment]
This has been implemented in 98b7e1de. Using --applet with --apdu will first select the applet AID.
